### PR TITLE
Migrate to gradle

### PR DESCRIPTION
**Pull request by sputnik27**
This PR introduces a gradle-based build. It contains also the commit that introduces user documentation, but since the documentation depends on the used build tool, I consider this commit also part of the migration.

### CODE REVIEW

**Comment by sputnik27** [comment]
@tkrille What do you think about this? Do you see any reason why this should not be merged?

**Comment by tkrille** [comment]
Unfortunately, I'm not so familiar with Gradle. I've assigned this to @wallner. At least he knows more about Gradle than I do :)

File: README.md [hunk 1]
```diff
@@ -1,4 +1,36 @@
...
+
+## Building
+
+This project uses gradle, so you can build it using
+
+    ./gradlew
+
+## Running
+
+This guide assumes, that the [OSIAM docker](https://github.com/osiam/docker-image) image is running.
```
Comment (wallner) [inline README.md]: The comma is wrong

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+jar {
+    manifest {
+        attributes  \
+             'Main-Class': mainClassName
+    }
+}
+
+dependencies {
+    compile(
+            'org.osiam:connector4java:1.7-SNAPSHOT',
```
Comment (wallner) [inline build.gradle]: Please extract the version into extra properties.

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            'de.raysha.lib:jsimpleshell:3.0.2'
```
Comment (wallner) [inline build.gradle]: ditto

**Comment by wallner** [comment]
There are two comma in the commit message, that don't belong there. Please remove them. My suggestion would be:

    This commit introduces a very basic README-file that is not complete yet. The README should provide
    all the information necessary to use the application for the most common use cases.

**Comment by wallner** [comment]
Do you have a plan how you would like to tag- and push releases to sonartype with gradle yet?

**Comment by sputnik27** [comment]
To answer your question on pushing artifacts: Not yet. I did some google searches on that topic, but did not yet find a plugin that fits the purpose. I don't consider this a blocker issue, because we will not release a version of the shell in the very near future.

In general, we need a good solution for this for all the other projects as well. I will investigate this.

**Comment by wallner** [comment]
I think the easiest way would be to migrate to [bintray](https://www.bintray.com). The documentation there is fairly comprehensive documentation and we could use [oss.jfrog.org] to publish snapshots. Maybe this repository is the ideal candidate to try out a workflow I will create a new issue for that.

**Comment by wallner** [comment]
And I will merge this PR. Thank you for your effort.

**Pull request merged**